paddle/fluid/operators/fused_elemwise_activation_op.cc

```diff
@@ -59,29 +114,42 @@ class FusedElemwiseActivationOp : public framework::OperatorWithKernel {
 class FusedElemwiseActivationMaker : public framework::OpProtoAndCheckerMaker {
  public:
   void Make() override {
-    AddInput("X", "(vector<Tensor>)");
-    AddInput("Y", "(vector<Tensor>)");
-    AddOutput("Out", "vector<Tensor>");
+    AddInput(
+        "X",
+        "(Tensor) The input tensor of fused_elemwise_activation operator.");
+    AddInput(
+        "Y",
+        "(Tensor) The input tensor of fused_elemwise_activation operator.");
+    AddOutput(
+        "Out",
+        "vector<Tensor> The output tensor of fused_elemwise_activation "
+        "operator. If the option of 'keep_intermediate_value' is opened, the "
+        "'Out' should contain 'out' and 'intermediate_out'."
+        "else, the 'Out' only contain 'out'.")
+        .AsDuplicable();
     AddAttr<int>("axis",
                  "axis is used by elementwise_op, the default value is -1.")
         .SetDefault(-1);
     AddAttr<float>("scale",
                    "scale is used by scale_op, the default value is 0.0.")
         .SetDefault(0.0);
-    AddAttr<bool>("recomputation",
-                  "Whether to recompute the Out."
-                  "fused_elemwise_activation_grad has two methods to get the "
-                  "dx and dy, one "
-                  "is to use the 'Out', and the other is not to use it. "
-                  "The former method will save the time of recomputing the "
-                  "'Out', but it must occupy the memory to store the 'out'. "
-                  "While, the later method can avoid occupying the memory, "
-                  "but it must recompute the 'Out'. The default value is true.")
+    AddAttr<bool>(
+        "recomputation",
```

Comment:
I assume this the data flow analysis graph's "edge" is decided by this attr? Will it be possible to detect the edge by memory_optimizer through attr?

Perhaps, a better way in the future is to have "FusedElemwiseActivation" and "FusedElemwiseActivationWithOut". They have different operator outputs, and there fore explicit "edge" for dependency analysis

Reply:
> Will it be possible to detect the edge by memory_optimizer through attr?

Maybe the memory_optimizer can't be finished by only one pass, so there may be one pass can detect whether some backward_op can support recomputation to get the `out` of the corresponding forward_op.

> Perhaps, a better way in the future is to have "FusedElemwiseActivation" and "FusedElemwiseActivationWithOut". They have different operator outputs, and there fore explicit "edge" for dependency analysis

Maybe.